src/app/features/ledger/components/ledger-inline-warnings.tsx

```diff
@@ -49,14 +61,14 @@ export function CommonLedgerDeviceInlineWarnings({
   chain,
   latestDeviceResponse,
   outdatedLedgerAppWarning = false,
+  incorrectAppOpened,
 }: CommonLedgerInlineWarningsProps) {
-  if (!latestDeviceResponse) return null;
-
   if (outdatedLedgerAppWarning) {
     return <OutdatedLedgerAppWarning chain={chain} />;
   }
-  if (latestDeviceResponse.deviceLocked) return <LedgerDeviceLockedWarning chain={chain} />;
-  if (isStacksLedgerAppClosed(latestDeviceResponse))
+  if (latestDeviceResponse?.deviceLocked) return <LedgerDeviceLockedWarning chain={chain} />;
+  if (incorrectAppOpened) return <LedgerIncorrectAppWarning chain={chain} />;
+  if (latestDeviceResponse && isStacksLedgerAppClosed(latestDeviceResponse))
```

Comment:
No change needed now, but would love to find a better approach to these common errors than this component. This was written when we just had stacks support and doesn't scale well for other assets types. Perhaps route state so these can just be navigated too 🤷🏼